src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/resources/gtk.properties

```
@@ -59,9 +59,9 @@ FileChooser.newFolderNoDirectoryErrorTitle.textAndMnemonic=Error
 FileChooser.newFolderNoDirectoryError.textAndMnemonic=Error creating directory "{0}": No such file or directory
 FileChooser.deleteFileButton.textAndMnemonic=De&lete File
 FileChooser.renameFileButton.textAndMnemonic=&Rename File
-FileChooser.cancelButton.textAndMnemonic=Cancel
+FileChooser.cancelButton.textAndMnemonic=&Cancel
 FileChooser.saveButton.textAndMnemonic=Save
```

```suggestion
FileChooser.saveButton.textAndMnemonic=&Save
```
It looks like the Save button is the only button left in this file without a mnemonic specified.

---

Hmm, the `&S` is already used by 
```
FileChooser.pathLabel.textAndMnemonic=&Selection:
```

It looks like that `&S` is more common for the save action, so probably we might want to change the `&Selection:` to something else, e.g. `S&election:`
Any thoughts?

---

mnemonics if gtk dialog were updated by the https://bugs.openjdk.org/browse/JDK-8080628, the save button was not changes since it is a default button in that dialog.

---

> mnemonics if gtk dialog were updated by the https://bugs.openjdk.org/browse/JDK-8080628, the save button was not changes since it is a default button in that dialog.

Seems like the fix is to remove mnemonics for Metal Look and Feel. 
Mnemonics for Open, Save and Cancel buttons are present on native side (checked with gedit application on ubuntu 22.04). So, I think mnemonics for Save button should also be there in GTK L&F.

---

> Hmm, the `&S` is already used by
> 
> ```
> FileChooser.pathLabel.textAndMnemonic=&Selection:
> ```
> 
> It looks like that `&S` is more common for the save action, so probably we might want to change the `&Selection:` to something else, e.g. `S&election:` Any thoughts?

Checked in native side and `&S` is the mnemonics for Save button. So, I think it's right to have `&S` as mnemonics for `Save` and as you suggested `&e` for `Selection`.

---

JDK-8080628 removes these Save and Open mnemonics from gtk and it does from all locales, not only the default locale, I guess for some customer escalation, so it needs to be verified with the fix author if it will regress the expectation.

If you want to reinstate the mnemonics, then I guess it needs to be reinstated for all locales gtk_de.properties, gtk_es.properties, gtk_fr.properties etc simultaneously

---

> JDK-8080628 removes these Save and Open mnemonics from gtk and it does from all locales, not only the default locale, I guess for some customer escalation, so it needs to be verified with the fix author if it will regress the expectation.

Ok.

> If you want to reinstate the mnemonics, then I guess it needs to be reinstated for all locales gtk_de.properties, gtk_es.properties, gtk_fr.properties etc simultaneously


I guess only `gtk.properties` needs to update and other locale should be taken care by g11n process as mentioned in one of my previous [PR](https://github.com/openjdk/jdk/pull/9327#discussion_r911212025).

---

Not sure if it still valid as I guess team doesn't exist anymore and probably Damon is doing the translations now but you can check..I brought it up since JDK-8080628 remove under one PR...so even if g11n process does it, it needs to be done offline and placed under this PR, I presume..

---

Ok.. I will check with Damon and if it is required to update other locales, will update.

---

I guess this will not require g11 process as there is no new translations needed as such, you just need to add the "&" back which was removed in JDK-8080628 but 
more important is JDK-8080628 has the change in all trains starting from jdk8 so if you are reverting gtk from it, I guess it needs to be done/backported for all those trains, which is why understanding why JDK-8080628 was done from the fix author is important..
Is Filechooser shown in GTK L&F **same** as what is shown in native file filechooser in ubuntu?

---

>I guess this will not require g11 process as there is no new translations needed as such, you just need to add the "&" back which was removed in JDK-8080628 but
more important is JDK-8080628 has the change in all trains starting from jdk8 so if you are reverting gtk from it, I guess it needs to be done/backported for all those trains, which is why understanding why JDK-8080628 was done from the fix author is important..

Sure, I will check with the author of JDK-8080628.

> Is Filechooser shown in GTK L&F same as what is shown in native file filechooser in ubuntu?

Overall UI is different in GTK and native filrchooser but the mnemonics are available for open/ cancel/ save buttons.